app/coffee/Features/Compile/ClsiManager.coffee

```diff
@@ -101,56 +118,101 @@ module.exports = ClsiManager =
 		return outputFiles
 
 	VALID_COMPILERS: ["pdflatex", "latex", "xelatex", "lualatex"]
+
 	_buildRequest: (project_id, options={}, callback = (error, request) ->) ->
-		Project.findById project_id, {compiler: 1, rootDoc_id: 1, imageName: 1}, (error, project) ->
+		Project.findById project_id, {}, (error, project) ->
```

Comment:
We may want to look at doing this with mongojs rather than mongoose to avoid the ORM overhead.